bullet-featherstone/src/SDFFeatures.cc

```diff
@@ -1051,34 +1051,92 @@ bool SDFFeatures::AddSdfCollision(
          ++submeshIdx)
     {
       auto s = mesh->SubMeshByIndex(submeshIdx).lock();
-      auto vertexCount = s->VertexCount();
-      auto indexCount = s->IndexCount();
-      btAlignedObjectArray<btVector3> convertedVerts;
-      convertedVerts.reserve(static_cast<int>(vertexCount));
-      for (unsigned int i = 0; i < vertexCount; i++)
+      bool meshCreated = false;
+      if (meshSdf->Optimization() ==
+          ::sdf::MeshOptimization::CONVEX_DECOMPOSITION ||
+          meshSdf->Optimization() ==
+          ::sdf::MeshOptimization::CONVEX_HULL)
```

Comment:
consider moving this logic outside of the SubMesh `for` loop

Reply:
moved out of `for` loop. [1f2aaa7](https://github.com/gazebosim/gz-physics/pull/606/commits/1f2aaa7a24756c51f56b5d03ad56958926d38630)